uint32 task_counter 
---
bool success